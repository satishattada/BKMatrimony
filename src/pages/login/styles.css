.login-container {
     margin: 20px; 
    /* height: 100vh; */
    margin: 10px;
}

.main-content {
    margin: 20px;
    border: 2px solid #eee;
    border-radius: 5px;
    padding: 20px;
    width: 100%;
    height: 368px
}
.login-content {
    float: right;
    justify-content: center;
    align-items: center;
    /* height: 100vh; */
    width: 100%;
}

.wedding-image {
    
    width: 80%;
}

.submit {
    margin-top: 20px;
    padding: 5px 50px;
    background: rgb(237, 119, 119);
    border: 1px solid red;
    border-radius: 4px;
    margin-left: 80px;

}