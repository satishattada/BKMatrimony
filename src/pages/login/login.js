
import React, { Component } from "react";
import './styles.css';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as appAction from '../../redux/actions';
import mar from './../../assets/mar3.jpg';
import Input from '../../components/input/input';
import Button from '../../components/button/button';
import { Link } from "react-router-dom";

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      errors: {

        username: '',
        password: '',

      }
    }
  }

  setUserName = (username) => {
    this.setState({ username })
  }

  setPassword = (password) => {
    this.setState({ password })
  }

  validateForm = () => {
    const { username, password } = this.state
    let validation = true;
    if (username.length === 0) {
      validation = false;
    }
    if (password.length === 0) {
      validation = false;
    }
    return validation;
  }

  handleSubmit = () => {
    const payload = {
      email: this.state.username,
      password: this.state.password
    }
    const validationStatus = this.validateForm()
    // if (validationStatus === false) {
    //   alert('please fill  the required fields')
    // }
    this.props.appAction.loginUser(payload);
  }

  render() {
    const { errors } = this.state;
    return (
      <div className="login-container">
        <div className="row m-0">
          <div className="col-md-6">
            <img className="wedding-image" src={mar} alt="wedding  logo" />
          </div>
          <div className="col-md-5">
            <div className="login-content">
              <div className="main-content">
                <h1>Login</h1>
                <Input 
                  testID="userNameTestId"
                  labelName="User Name"
                  type="text"
                  placeholder="User Name" onChangeEvent={(event) => { this.setUserName(event.target.value) }} />
                <label className="errorStyle">{errors.username}</label>

                <Input 
                 testID="passwordTestId"
                  labelName="Password"
                  type="password"
                  placeholder="Password" onChangeEvent={(event) => { this.setPassword(event.target.value) }} />
                <label className="errorStyle">{errors.password}</label>

                <Button testID="submitTestID" btnClass="btn-warning" value="Login" onSubmitBtn={this.handleSubmit} />

                <div className="rplinks">
                  <div className="row">
                    <div className="col-md-9">
                      <Link to={`/forgot-password`}>{'ForgotPassword'}</Link>
                    </div>
                    <div className="col-md-3">
                      <Link to={`/register`}>{'Register'}</Link>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>

      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  appAction: bindActionCreators(appAction, dispatch),
});

export const mapStateToProps = (state) => {
  return {
    userData: state.user.userData
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);

