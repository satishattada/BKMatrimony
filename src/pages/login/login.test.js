import toJson from 'enzyme-to-json';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Login, mapDispatchToProps, mapStateToProps } from './login';
let wrapper;

configure({ adapter: new Adapter() });
const defaultProps = {
  appAction: {
    loginUser: jest.fn().mockImplementation(() => Promise.resolve()),
  },
  navigate: jest.fn().mockImplementation('/register')

}
beforeEach(() => {
  wrapper = shallow(<Login {...defaultProps} />);
});

describe('<Login />', () => {
  it('should render App component without crashing', () => {
    expect(wrapper).toBeTruthy();
  });

  it('<AdaptiveTaskInstructions /> Snapshot Test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('username test', () => {
    const component = wrapper.instance();
    const uName = wrapper.find({testID:'userNameTestId'}).props();
    uName.onChangeEvent({target: {value: 'satish@gmail.com'}});
    expect(component.state.username).toEqual('satish@gmail.com');
  });

  it('username test not equal', () => {
    const component = wrapper.instance();
    const uName = wrapper.find({testID:'userNameTestId'}).props();
    uName.onChangeEvent({target: {value: 'satish@gmail.com'}});
    expect(component.state.username).not.toBe('aparna@gmail.com');
  });

  it('password test', () => {
    const component = wrapper.instance();
    const uName = wrapper.find({testID:'passwordTestId'}).props();
    uName.onChangeEvent({target: {value: 'attada'}});
    expect(component.state.password).toEqual('attada');
  });

  it('submit button test', () => {
    const uName = wrapper.find({testID:'submitTestID'}).props();
    uName.onSubmitBtn();
  });

  
  it('register button test', () => {
    const uName = wrapper.find({testID:'registerTestID'}).props();
    uName.onSubmitBtn();
  });

  it('validate form test', () => {
    const component = wrapper.instance();
    wrapper.setState({username: 'satish', password: 'attada'});

    const result = component.validateForm();
    expect(result).toBeTruthy();
    // expect(component.state.password).toEqual('attada');
  });

  it('validate form test with username empty', () => {
    const component = wrapper.instance();
    wrapper.setState({username: '', password: 'attada'});

    const result = component.validateForm();
    expect(result).toBeFalsy();
    // expect(component.state.password).toEqual('attada');
  });

  it('validate form test with password empty', () => {
    const component = wrapper.instance();
    wrapper.setState({username: 'satisg', password: ''});

    const result = component.validateForm();
    expect(result).toBeFalsy();
    // expect(component.state.password).toEqual('attada');
  });

  it('should call mapState', () => {
    const mockedState = {
      user: {
        userData: {
        name: 'aaaa',
        email: 'xxx@gmail.com',
      }}
    };
    const state = mapStateToProps(mockedState);
    expect(state.userData).toBeDefined();
  });

  it('should call mapDispatchToProps', () => {
    const mockedState = {};
    const state = mapDispatchToProps(mockedState);
    expect(state.appAction).toBeDefined();
  });


});    