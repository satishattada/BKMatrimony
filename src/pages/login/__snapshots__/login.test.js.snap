// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> <AdaptiveTaskInstructions /> Snapshot Test 1`] = `
<div
  className="container"
>
  <div
    className="login-container"
  >
    <div
      className="row m-0"
    >
      <div
        className="col-md-5"
      >
        <img
          alt="wedding  logo"
          className="wedding-image"
          src="mar3.jpg"
        />
      </div>
      <div
        className="col-md-6"
      >
        <div
          className="login-content"
        >
          <div
            className="main-content"
          >
            <h1
              className="lheading"
            >
              Login
            </h1>
            <Input
              labelName="User Name"
              onChangeEvent={[Function]}
              placeholder="User Name"
              testID="userNameTestId"
              type="text"
            />
            <label
              className="errorStyle"
            />
            <Input
              labelName="Password"
              onChangeEvent={[Function]}
              placeholder="Password"
              testID="passwordTestId"
              type="password"
            />
            <label
              className="errorStyle"
            />
            <Button
              btnClass="btn-warning"
              onSubmitBtn={[Function]}
              testID="submitTestID"
              value="Login"
            />
            <div
              className="fpli"
            >
              <Link
                to="/forgot-password"
              >
                ForgotPassword
                 ?
              </Link>
            </div>
            <p
              className="dheading"
            >
              Don't have an account?
            </p>
            <Button
              btnClass="btn-danger"
              onSubmitBtn={[Function]}
              testID="registerTestID"
              value="Register"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
